ops/book/src/ci/pr-authorization.md

```diff
@@ -0,0 +1,57 @@
+# PR Authorization
+
+Jobs running on self-hosted infrastructure will not run on forked PRs. This is a hard limitation of
+CircleCI. Additionally, any jobs that use secrets or contexts will not run if the PR's owner is not part of the
+`ethereum-optimism` GitHub org. This is a security policy: without these requirements, anyone could open a forked PR
+and run untrusted code on our executors, or exfiltrate secrets.
+
+To get around this, forked PRs must be authorized by pushing them to the main repository. We run a bot called
+[bailiff][bailiff] to do this.
+
+To use Bailiff, you must be a member of the `engineering` team on GitHub. Then, to authorize a PR, post a comment
+that looks like this:
+
+```shell
+/ci authorize <full commit hash>
+
+- or -
+
+/ci authorize <full URL to commit>
+```
+
+The commit hash is compared with the head of the forked PR. If they don't match, Bailiff will refuse to push the PR.
+This prevents malicious users from ninja-pushing to PRs after they've been authorized, and ensures that we know
+exactly what is running on our infrastructure. **New commits will have to be authorized again.**
+
+Note: Bailiff does not push to the source branch on the fork. It pushes to its own branch based on a hash of the source
+branch. It'll look like `external-fork/<sha256 hash>`. For this reason, **re-running CI jobs on the source branch
+will not work.** Instead, re-run the commit on the `external-fork` branch or push a new commit and re-authorize.
```

Comment:
Just wondering, does Bailiff have an async routine of cleaning up those branches on the upstream / source branch? (otherwise, one could flood them maliciously)

Reply:
it does not - that's something we'll have to add